daemon/daemon.go

```diff
@@ -511,6 +530,20 @@ var (
 	rebootMaxTentatives    = 3
 )
 
+func (d *Daemon) writeMaintenanceFile(rst state.RestartType) error {
+	b, err := json.Marshal(maintenanceForRestartType(rst))
+	if err != nil {
+		return err
+	}
+
+	err = ioutil.WriteFile(dirs.SnapdMaintenanceFile, b, 0644)
```

Comment:
Didn't read the rest so maybe a silly question: do we have to create the directory?

Reply:
I guess maybe but the directory `/var/lib/snapd` should exist in order for daemon to do things anyways? Said another way, I'm not sure that it makes sense for snapd the daemon to be able to run if `/var/lib/snapd/` doesn't exist?

Reply:
this should probably be an AtomicWrite no?